packages/flutter/lib/src/animation/tween.dart

```diff
@@ -440,15 +521,37 @@ class CurveTween extends Animatable<double> {
   /// The curve to use when transforming the value of the animation.
   Curve curve;
 
+  @override
+  double get begin => 0.0;
+
+  @override
+  double get end => 1.0;
+
+  @override
+  double lerp(double t) => curve.transform(t);
+
   @override
   double transform(double t) {
     if (t == 0.0 || t == 1.0) {
       assert(curve.transform(t).round() == t);
       return t;
     }
-    return curve.transform(t);
+    return super.transform(t);
   }
 
   @override
   String toString() => '${objectRuntimeType(this, 'CurveTween')}(curve: $curve)';
 }
+
+
+mixin TweenWithoutSettersMixin<T> on Tween<T> {
```

Comment:
This also seems the wrong way around. Instead of disallowing calling the setters, there should be a base class that doesn't have them.